### PR TITLE
Handle more text editing intents

### LINKED ISSUE #180: Some text editing intents are not implemented
The following intents need to be implemented:

- `moveToLeftEndOfLine`: ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: true),
- `moveToRightEndOfLine`: ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: true),
- `moveToBeginningOfDocument`: ExtendSelectionToDocumentBoundaryIntent(forward: false, collapseSelection: true),
- `moveToEndOfDocument`: ExtendSelectionToDocumentBoundaryIntent(forward: true, collapseSelection: true),
- `moveToLeftEndOfLineAndModifySelection`: ExpandSelectionToLineBreakIntent(forward: false),
- `moveToRightEndOfLineAndModifySelection`: ExpandSelectionToLineBreakIntent(forward: true),
- `moveToBeginningOfDocumentAndModifySelection`: ExpandSelectionToDocumentBoundaryIntent(forward: false),
- `moveToEndOfDocumentAndModifySelection`: ExpandSelectionToDocumentBoundaryIntent(forward: true),
- `scrollToBeginningOfDocument`: ScrollToDocumentBoundaryIntent(forward: false),
- `scrollToEndOfDocument`: ScrollToDocumentBoundaryIntent(forward: true),
- `scrollPageUp`: ScrollIntent(direction: AxisDirection.up, type: ScrollIncrementType.page),
- `scrollPageDown`: ScrollIntent(direction: AxisDirection.down, type: ScrollIncrementType.page),
- `pageUpAndModifySelection`: ExtendSelectionVerticallyToAdjacentPageIntent(forward: false, collapseSelection: false),
- `pageDownAndModifySelection`: ExtendSelectionVerticallyToAdjacentPageIntent(forward: true, collapseSelection: false),

### PR DESCRIPTION
**Pull request by amantoux**
Closes #180

### CODE REVIEW

**Comment by amantoux** [comment]
@Amir-P can you please review?

File: packages/fleather/lib/src/widgets/editor.dart [hunk 1]
```diff
@@ -910,8 +912,28 @@ class RawEditorState extends EditorState
       ),
       cause,
     );
+
     if (cause == SelectionChangedCause.toolbar) {
-      bringIntoView(textEditingValue.selection.extent);
+      switch (defaultTargetPlatform) {
```
Comment (Amir-P) [inline packages/fleather/lib/src/widgets/editor.dart:917]: Isn't this weird? On master we have this change but it compares it with the old version. @amantoux

**Review by Amir-P — APPROVED** [review]
LGTM. Just can you check the comment? @amantoux

File: packages/fleather/lib/src/widgets/editor.dart (hunk 1 above)
Comment (amantoux) [reply packages/fleather/lib/src/widgets/editor.dart:917]: I fail to understand to what you mean by 'it compares it with the old version'?

**Pull request merged**